controller/src/main/java/io/pravega/controller/store/stream/ZKStream.java

```diff
@@ -110,23 +107,58 @@ public void refresh() {
     }
 
     @Override
-    public CompletableFuture<Void> checkStreamExists(final Create create) {
+    public CompletableFuture<CreateStreamResponse> checkStreamExists(final StreamConfiguration configuration, final long creationTime) {
+        // If stream exists, but is in a partially complete state, then fetch its creation time and configuration and any metadata that is available from a previous run
+        // If the existing stream has already been created successfully earlier,
         return store.checkExists(creationPath)
-                .thenCompose(x -> {
-                    if (x) {
-                        return cache.getCachedData(creationPath)
-                                .thenApply(creationTime -> BitConverter.readLong(creationTime.getData(), 0) != create.getCreationTime());
+                .thenCompose(exists -> {
+                    if (exists) {
```

Comment:
If you invert this if, you can remove the else block (remove the keyworkd and indentation but keep the content).
This would avoid this being indeed so far and having to wrap so often. The same could be done for the configExists if below.